app/models/student.rb

```diff
@@ -14,7 +14,11 @@ class Student < ActiveRecord::Base
   has_many :event_notes, dependent: :destroy
   has_many :services, dependent: :destroy
   has_many :tardies, dependent: :destroy
+  has_many :dashboard_tardies, -> { where('occurred_at >= ?', 3.months.ago) },
```

Comment:
@edavidsonsawyer I'm curious about the different scopes on tardies versus absences here. Also wondering if this is something the student model needs to be concerned about, if I want to change the schoolwide absences dashboard I think I'd be surprised if I needed to change `student.rb`.

Reply:
This deserves some discussion, yeah. The problem is, if we send all the attendance records to the dashboard, filtering that to a sensible range becomes slow, especially since the dashboard isn't restricted to active students. I had thought I could handle that in the school controller by restricting the included records, but it turns out that approach didn't work. Unless there's something I'm missing, you can't only bring in a subset of records with includes. 

I'm open to alternatives! This was the only way I come up with to get the range of dates the dashboard needs to load initially without running into performance problems. Conceptually I'm thinking of these as a student's "current" attendance data, but I'm also not wild about this approach.

Reply:
@edavidsonsawyer Is this helpful?

http://guides.rubyonrails.org/active_record_querying.html#specifying-conditions-on-eager-loaded-associations

Rails guides example:

```
Article.includes(:comments).where(comments: { visible: true })
```

Reply:
@alexsoble Yeah, that's what I tried initially, but it doesn't work the way I thought it did (and consequently the current dashboard is doing this incorrectly). If you look at the query that generates, it doesn't filter the included records, but applies the condition to the joined tables. So in our case it would return only those students with absences in the timeframe. That's no good because we do want the students with no absences w/in the last year, we just don't want their absences. Basically, it excludes students with perfect attendance within the timeframe.

Reply:
@edavidsonsawyer Ah, OK, that's some good tech learning! My only suggestion is to think about using a more specific variable name if possible. But I don't think that should be a blocker to merging in this work.